internal/ostree/ostree.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ostree
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"syscall"
+
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	rpmostreeclient "github.com/openshift-kni/lifecycle-agent/ibu-imager/ostreeclient"
+)
+
+// TODO: replace with https://github.com/coreos/rpmostree-client-go
+//       when https://github.com/coreos/rpmostree-client-go/issues/23 resolves
+
+// QueryStatus query the status of rpm-ostree
+func QueryStatus() (*rpmostreeclient.Status, error) {
+	cmd := exec.Command("/usr/bin/env", "--", "bash", "-c", "rpm-ostree status --json")
+	cmd.SysProcAttr = &syscall.SysProcAttr{Chroot: utils.Host}
+	var stdoutBytes, stderrBytes bytes.Buffer
+	cmd.Stdout = &stdoutBytes
+	cmd.Stderr = &stderrBytes
+	cmd.Dir = "/"
+	err := cmd.Run()
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", stderrBytes.String(), err)
+	}
+	var q rpmostreeclient.Status
+	if err := json.Unmarshal(stdoutBytes.Bytes(), &q); err != nil {
+		return nil, fmt.Errorf("failed to parse `rpm-ostree status --json` output: %w", err)
```

Comment:
most likely you have checked...but are we intentionally skipping https://github.com/coreos/rpmostree-client-go ?

Reply:
updated